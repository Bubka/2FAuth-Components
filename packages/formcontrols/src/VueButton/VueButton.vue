<script setup>
    const props = defineProps({
        label: {
            type: String,
            default: '[[label]]'
        },
        color: {
            type: String,
            default: 'is-link'
        },
        isLoading: {
            type: Boolean,
            default: false
        },
        isDisabled: {
            type: Boolean,
            default: false
        },
        nativeType: {
            type: String,
            default: 'submit',
            required: true
        }
    })
</script>

<template>
    <button 
        :type="nativeType"
        :disabled="isLoading || isDisabled"
        :class="{
            'button': true,
            [`${color}`]: true,
            'is-loading': isLoading,
        }">
        <slot>
            {{ $t(props.label) }}
        </slot>
    </button>
</template>