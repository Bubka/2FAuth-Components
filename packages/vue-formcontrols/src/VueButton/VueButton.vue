<script setup>
    const props = defineProps({
        color: {
            type: String,
            default: 'is-link'
        },
        isLoading: {
            type: Boolean,
            default: false
        },
        isDisabled: {
            type: Boolean,
            default: false
        }
    })
</script>

<template>
    <button 
        type="button"
        :disabled="isLoading || isDisabled"
        :class="{
            'button': true,
            [`${color}`]: true,
            'is-loading': isLoading,
        }">
        <slot />
    </button>
</template>