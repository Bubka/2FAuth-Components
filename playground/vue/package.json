{
  "name": "vue",
  "private": true,
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "vue-tsc -b && vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@2fauth/styles": "link:../../packages/styles",
    "@2fauth/formcontrols": "link:../../packages/formcontrols",
    "vue": "^3.5.13"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^5.2.1",
    "@vue/tsconfig": "^0.7.0",
    "sass": "1.67.0",
    "typescript": "~5.7.2",
    "vite": "^6.2.0",
    "vue-tsc": "^2.2.4"
  }
}
