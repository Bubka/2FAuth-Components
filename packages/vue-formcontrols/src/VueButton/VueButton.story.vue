<script lang="ts" setup>
    import { logEvent } from 'histoire/client'
    import VueButton from './VueButton.vue'

    function initState() {
        return {
            label: 'Label',
        }
    }
</script>

<template>
    <Story :init-state="initState">
        <template #default="{ state }">
            <VueButton :label="state.label" @click="logEvent('Click',$event)" />
        </template>
    </Story>
</template>