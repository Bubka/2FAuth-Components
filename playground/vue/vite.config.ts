import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import VueI18nPlugin from '@intlify/unplugin-vue-i18n/vite'
import path from 'path'

// https://vite.dev/config/
export default defineConfig({
    plugins: [
        vue(),
        VueI18nPlugin({
            include: [path.resolve(__dirname, '../../resources/lang/*.json')],

        })
    ],
})
