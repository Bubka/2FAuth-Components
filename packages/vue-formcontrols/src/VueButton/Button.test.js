import { fireEvent, render, screen } from '@testing-library/vue'
import { describe, expect, it } from 'vitest'
import VueButton from './VueButton.vue'

describe('button', () => {
    it('should emit a click event when clicked', async () => {
        const { emitted } = render(VueButton, {
            props: {
                label: 'label',
            },
        })

        await fireEvent.click(screen.getByText('label'))

        expect(emitted()).toMatchInlineSnapshot(`
      {
        "click": [
          [
            MouseEvent {
              "isTrusted": false,
            },
          ],
        ],
      }
    `)
    })

    describe('color', () => {
        it('should render a is-link button by default', () => {
            const button = render(VueButton)

            expect(button.html()).toMatchSnapshot()
        })

        it('should render a custom color button', () => {
            const button = render(VueButton, {
                props: {
                    color: 'is-success',
                },
            })

            expect(button.html()).toMatchSnapshot()
        })
    })

    describe('label', () => {
        it('should render a default label with bracket', () => {
            const button = render(VueButton)

            expect(button.html()).toMatchSnapshot()
        })

        it('should render passed label', () => {
            const button = render(VueButton, {
                props: {
                    label: 'label',
                },
            })

            expect(button.html()).toMatchSnapshot()
        })
    })
})