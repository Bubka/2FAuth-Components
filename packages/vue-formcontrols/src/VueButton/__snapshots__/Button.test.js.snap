// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > color > should render a custom color button 1`] = `"<button type="button" class="button is-success">[[label]]</button>"`;

exports[`button > color > should render a is-link button by default 1`] = `"<button type="button" class="button is-link">[[label]]</button>"`;

exports[`button > label > should render a default label with bracket 1`] = `"<button type="button" class="button is-link">[[label]]</button>"`;

exports[`button > label > should render passed label 1`] = `"<button type="button" class="button is-link">label</button>"`;
